### PR TITLE
Implemented link following using numbers

### PR DESCRIPTION
**Pull request by shinglyu**
r? @johnathan79717

### CODE REVIEW

**Comment by shinglyu** [comment]
I'll merge #36 first then rebase this one

**Comment by shinglyu** [comment]
Rebased, please review, thanks.

File: vim.js [hunk 1]
```diff
@@ -181,3 +156,94 @@ function copyCurrentLocation() {
...
+    codehint.style.border="solid 1px black";
+    codehint.style.backgroundColor="white";
+    codehint.style.font="12px/14px bold sans-serif";
+    codehint.style.color="darkred";
+    codehint.style.position="absolute";
+    codehint.style.top="0";
+    codehint.style.left="0";
+    codehint.style.padding="0.1em";
+
+    elem.style.position="relative";
```
Comment (johnathan79717) [inline vim.js:182]: We changed elem.style.position here. Do we need to recover it in reduce_highlights()?

File: vim.js [hunk 2]
```diff
@@ -145,6 +119,7 @@ document.addEventListener('keypress', function(evt){
   console.log("State after: " + gState);
 });
 
+
```
Comment (johnathan79717) [inline vim.js]: Is this newline necessary?

**Comment by johnathan79717** [comment]
Well done!
r=me if the above comments are answered.

File: vim.js [hunk 3]
```diff
@@ -181,3 +155,94 @@ function copyCurrentLocation() {
...
+/* Link Following */
+function highlight_links() {
+  // TODO: buttons, inputs
+  var links = document.querySelectorAll('a');
+  // TODO: asdfghjkl; codes
+  var code = 0;
+  Array.prototype.forEach.call(links, function(elem){
+    // console.log(elem);
+    elem._originalBackgroundColor = elem.style.backgroundColor;
+    elem._originalPosition = elem.style.position;
```
Comment (shinglyu) [inline vim.js:168]: I save the original style directly in the DOM element

**Pull request merged**